### PR TITLE
Rework Vimrc with some usefull functionality and nice comments

### PR DESCRIPTION
**Pull request by DanielSchuerhoff**

### CODE REVIEW

**Review by NOhs — COMMENTED** [review]
I can't test atm so I can only comment, that it looks fine to me.

File: configuration_zsh/.vimrc [hunk 1]
```diff
@@ -1,15 +1,51 @@
...
-set tabstop=4
-set shiftwidth=4
-set smartindent
-set smarttab
-set softtabstop=4
-set ruler
-set backspace=indent,eol,start
-set expandtab
+"=============Colors============= 
+colorscheme default		" colorscheme if you so wish
```
Comment (GPMueller) [inline configuration_zsh/.vimrc:2]: @NOhs did you have a colorscheme somewhere, which you wanted to use? Or is the default fine with our colors palette?

File: configuration_zsh/.vimrc [hunk 2]
```diff
@@ -1,15 +1,51 @@
...
+"=============Space and Tabs============= 
+set tabstop=4			" number of visual spaces per TAB
+set softtabstop=4		" number of spaces in tab when editing
+set shiftwidth=4		" number of spaces for autoindent
+set expandtab			" convert tabs into spaces
+set autoindent          " automatically indent a line like the one before
+filetype indent on		" load filetype-specific indent files
+"set smartindent         " nicer and smarter indentation
+                        " does not work with filetype indent
+set backspace=indent,eol,start      " Chance backspace to the usual (non-vim) behaviour
```
Comment (GPMueller) [inline configuration_zsh/.vimrc:15]: typo: should be `Change`

File: configuration_zsh/.vimrc [hunk 3]
```diff
@@ -1,15 +1,51 @@
...
+"=============UI Config============= 
+set number				" show line numbers
+set showcmd				" show command in bottom bar
+"set ruler               " enable a ruler, displaying line and column number
+                        " unnecessary since showcmd is enabled
+set cursorline			" highlight current line
+set showbreak=+++       " marking wrapped lines with +++
+set linebreak           " nicer line breaking/wrapping
+set wildmenu			" visual autocomplete for command menu
+set lazyredraw			" redraw only when necessary
```
Comment (GPMueller) [inline configuration_zsh/.vimrc:26]: what is this needed for? does it reduce latency or is it purely client-side?

**Review by GPMueller — COMMENTED** [review]
Looks good to me

**Review by GPMueller — APPROVED** [review]
This is a nice little improvement.

**Pull request merged**